### PR TITLE
`implode` and `explode`

### PR DESCRIPTION
**Pull request by tomekwi**
How about these?

### implode

Collapse a list of arguments into an array of arguments.

```js
var implode = require('1-liners/implode');

const f = (a, b) => a + b;

[
	[1, 2],
	[3, 4],
	[5, 6],
].map(implode(f));  // => [3, 7, 11]
```

### explode

The opposite of [implode](#implode).

```js
var explode = require('1-liners/explode');

const sum = (numbers) => numbers.reduce((a, b) => a + b);

explode(f)(1, 2, 3, 4);  // => 10
```

### CODE REVIEW

**Comment by stoeffel** [comment]
I will check this out tomorrow. But on a first blink it sounds great.

**Comment by stoeffel** [comment]
Isn't implode the same as apply? I would suggest to rename apply and add an uncurryied explode.

**Comment by tomekwi** [comment]
:+1: Right you are. I was thinking *apply* was badly named anyway. Because some day it’d be good to have *apply(context, f, args)* and *bind(context, f)*. *call* doesn’t make much sense as it doubles the functionality of apply, and doesn’t have a static number of args.

**Comment by stoeffel** [comment]
> Because some day it’d be good to have apply(context, f, args) and bind(context, f)

:+1: will you update your PR?

**Comment by tomekwi** [comment]
I’m at it right now. But now I got down to it I have second thoughts. Here’s what bothers me:

```js
test('#explode', () => {
	let f = implode((a) => a + a);
	let g = implode((a, b, c) => a * b * c);
	let h = implode((a, b) => a * b);

	equal(explode(f, 1), f([1]));
	equal(explode(g, 1, 2, 3), g([1, 2, 3]));
	equal(explode(h, 1, 2, 3), h([1, 2, 3]));
});
```

– *explode* wouldn’t have a stable number of arguments.

**Comment by stoeffel** [comment]
Yes, right. hmm... if we go with purefunctions. we should just skip it.

**Comment by stoeffel** [comment]
Or... I actually wouldn't mind to add functions with unstable number of args, if it is not possible otherwise.

**Comment by tomekwi** [comment]
I actually liked the `implode(f)` and `explode(f)` syntax. What `f` does and how many args it takes is none of our business. We just wrap the function – just like `curry` does.

**Comment by stoeffel** [comment]
> We just wrap the function – just like curry does.

tru. lets :ship: this

**Comment by tomekwi** [comment]
> I actually wouldn't mind to add functions with unstable number of args, if it is not possible otherwise.

Yeah, I also don’t. Anyway it’s very common in the JS world to overload functions and do defaults. Here’s what drove me to thinking it’d be good:

- a good selling point for the library – people buy bold statements
- it’s much easier to follow what’s happening when `curry`ing and `flip`ping

**Comment by tomekwi** [comment]
Try to wrap your head around what’s happening here :)

```js
import set from '101/set';
import curry from '1-liners/curry2';

curry(set)('prop') == ???
```

File: README.md [hunk 1]
```diff
@@ -292,14 +293,32 @@ flip(f)(2, 6);        // => 3
...
+### explode
+
+The opposite of [implode](#implode).
+
+```js
+var explode = require('1-liners/explode');
+
+const sum = (numbers) => numbers.reduce((a, b) => a + b);
+
+explode(f)(1, 2, 3, 4);  // => 10
```
Comment (stoeffel) [inline README.md]: `f` should be `sum`

**Comment by tomekwi** [comment]
:checkered_flag:  Sorry!

**Comment by stoeffel** [comment]
np, have a nice we:-)

**Comment by tomekwi** [comment]
:+1: ebenfalls!

**Pull request merged**